packages/core/src/components/menu/_common.scss

```diff
@@ -67,31 +75,40 @@ $dark-menu-item-color-hover: $dark-minimal-button-background-color-hover !defaul
       .pt-menu-item-label {
         color: map-get($pt-intent-colors, $intent);
       }
-    }
-  }
 
-  @each $intent, $color in $button-intents {
-    &.pt-intent-#{$intent} {
       &:hover,
       &.pt-active {
-        background: nth(map-get($button-intents, $intent), 1);
+        background-color: nth(map-get($button-intents, $intent), 1);
       }
 
       &:active {
-        background: nth(map-get($button-intents, $intent), 2);
+        background-color: nth(map-get($button-intents, $intent), 2);
       }
 
       &:hover,
       &:active,
       &.pt-active {
-        color: $white;
-
+        &,
         &::before,
         &::after,
         .pt-menu-item-label {
           color: $white;
         }
       }
+
+      &.pt-disabled {
+        background-color: inherit;
+
+        &,
+        &::before,
+        &::after {
+          color: $icon-color-disabled;
+        }
+
+        .pt-menu-item-label {
+          color: $text-color-disabled;
```

Comment:
this actually seems like an appropriate use for `!important`: disabled beats everything else, so it can force its styles to override more specific selectors. should allow you to revert most of this 👍